client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx

```diff
@@ -37,14 +31,21 @@ import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selector
 import { getProductBySlug, getProductsList } from 'calypso/state/products-list/selectors';
 import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
+import {
+	AddToCartAndCheckout,
+	PriceBadge,
+	PriceWithTerm,
+	recordTracksEventAddToCartClick,
+	TermLength,
+} from './utils';
 import type { EmailProvidersStackedCardProps, ProviderCard } from './provider-card-props';
 
 import './professional-email-card.scss';
 
 // eslint-disable-next-line @typescript-eslint/no-empty-function
 const noop = () => {};
 const logo = <Gridicon className="professional-email-card__logo" icon="my-sites" />;
-const badge = <img src={ poweredByTitanLogo } alt={ translate( 'Powered by Titan' ) } />;
+const badge = <img src={ poweredByTitanLogo } alt={ translate( 'Powered by Titan' ).toString() } />;
```

Comment:
alt attribute expects a string, not a TranslateResult. It was complaining I just make compiler happy 🙂

Reply:
That is strange. If I look up `alt={ translate(` in our code none of the occurrences call that function (even in `.tsx` files).

Reply:
Just to make sure I understand, what is complaining ? Os it PhpStorm? Or do you get errors when compiling or in your browser's console?

Reply:
![image](https://user-images.githubusercontent.com/5689927/145069605-d8c5c26a-7a90-4571-8117-b8eff3b84ff2.png)

Reply:
This is because `translate()` can return a React component in the case where you use component interpolation, so if `alt` requires a string, you have to cast the `TranslateResult` to a string first. You can also do this with `String( translate( '...' ) )`.

Reply:
Without the casting, this should throw a TS error in every case. For example, the error here: https://github.com/Automattic/wp-calypso/blob/89332041fdc9b23ea8b114cb1bbe840ad7e5d187/client/my-sites/marketing/tools/header.tsx#L18

and here:

https://github.com/Automattic/wp-calypso/blob/89332041fdc9b23ea8b114cb1bbe840ad7e5d187/client/components/jetpack/intro-pricing-banner/index.tsx#L32

Reply:
If the translation result must be a plain `string`, like in HTML attributes where HTML/JSX markup is not supported, you should declare it with the `textOnly` option:
```tsx
<img alt={ translate( 'Powered by Titan', { textOnly: true } ) as string } />
```
It's also better to cast `as string` instead of `.toString()`, because we only need to do a compile-time type cast for TypeScript, not a conversion at runtime.

#58938 improves the `translate` types to infer the `string` return type with `textOnly`. After that is merged, you won't need the `as string` cast.